File: swift/problem434.swift
class Solution {
    func countSegments(_ s: String) -> Int {
        var ans = 0; var prev = true
        let space = Character(" ")
        for char in s {
            if (char != space) {
                if prev {
                    ans += 1
                }
                prev = false
            } else {
                prev = true
            }
        }
        return ans
    }
}

File: java/problem434.java
class Solution {
    public int countSegments(String s) {
        if(s.length()==0)return 0;
        boolean tag=false;
        int result=0;
        for(int i=0;i<s.length();i++){
            if(!tag&&s.charAt(i)!=' '){
                tag=true;
            }
            if(tag&&(s.charAt(i)==' '||i==s.length()-1)){
                result++;
                tag=false;
            }
        }
        return result;
    }
}